custom_components/mammotion/entity.py

```diff
@@ -26,6 +26,7 @@ def __init__(
             identifiers={(DOMAIN, device_name)},
             manufacturer="Mammotion",
             serial_number=coordinator.device.luba_msg.net.toapp_wifi_iot_status.productkey,
+            sw_version=coordinator.device.luba_msg.net.toapp_devinfo_resp.resp_ids[0].info,
```

Comment:
Rsp_id in resp_ids

Reply:
@jLynx reminder to find a better way to get the value out than directly accessing the array.